core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -620,17 +623,67 @@ object ConditionalScriptPubKey extends ScriptFactory[ConditionalScriptPubKey] {
     fromAsm(asm)
   }
 
-  def isConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
+  def isConditionalScriptPubKeyWithElseIndex(
+      asm: Seq[ScriptToken]): (Boolean, Int) = {
     val headIsOpIf = asm.headOption.contains(OP_IF)
-    lazy val containsOpElse = {
-      val opElseIndex = asm.indexOf(OP_ELSE)
-      opElseIndex != -1
+    lazy val endsWithEndIf = asm.last == OP_ENDIF
+
+    var opElseIndexOpt: Option[Int] = None
+
+    lazy val validConditionalTree = {
```

Comment:
so first thought before reviewing this implementation: Can't we we use the `ConditionalCounter` some how validate this? From my first look, it seems we are just making sure that every `OP_IF` has a matching `OP_ENDIF`?

Reply:
We are also validating that there is only a single `OP_ELSE` at every level and that no asm appears after the top-level `OP_ENDIF`. For this reason, primarily the first one dealing with `OP_ELSE`, we cannot use `ConditionalCounter` as it would not be enough.